### PR TITLE
Fix broken links, convert some links to internal references

### PR DESCRIPTION
**Pull request by mechanomi**
fixing links first reported here: https://github.com/crate/tech-writing-domain/issues/336

for the purposes of making sure I got all the broken links in this PR, I manually patched the local build system so it has the fix in fix for https://github.com/crate/crate-docs/issues/30

### CODE REVIEW

**Comment by mechanomi** [comment]
I am currently blocked by the fact that the links to the Cloud pricing calculator are broken, and unfortunately, there is not a simple fix. the price calculator has been removed from the website so there is nothing to link to

I have contacted @matthijskrul and will pick this work up again once I am clear on how to resolve this issue

**Comment by mechanomi** [comment]
I will manually change the breaking links to this page locally so I can proceed with my testing. @matthijskrul has said that he will open a PR to fix the underlying issue

**Comment by mechanomi** [comment]
all links are now reported to work (with the exception of https://crate.io/products/cratedb-cloud/#cloud-calculator, which is going to be addressed by a separate PR, per my previous comment)

**Comment by mechanomi** [comment]
@matthijskrul can I get a review from you please? <3

File: docs/glossary.rst [hunk 1]
```diff
@@ -314,7 +314,8 @@ storage vs. more processing power, and so forth. They can also be further
 adjusted for different :ref:`scale units<glossary-scale-unit>` per plan.
 Currently there are four subscription plans available on our cloud offers, with
 more to come in the near future. For more information on choosing the right
-subscription plan, refer to our documentation `on the subject`_.
+subscription plan, refer to our :ref:`documentation on the subject
+<subscription-plans>`.
```
Comment (matthijskrul) [inline docs/glossary.rst:318]: Since we revised this now, can you get rid of this bit?

**Comment by mechanomi** [comment]
@matthijskrul pushed a fixup. can you please recheck? thx!

**Review by matthijskrul — APPROVED** [review]

**Pull request merged**